react/react1/week3/src/_app.jsx

```diff
@@ -0,0 +1,166 @@
+import { useEffect, useRef, useState } from "react";
+import "./App.css";
+function App() {
+  const [todos, setTodos] = useState([]);
+  const [counter, setCounter] = useState(1);
+  const [loading, setLoading] = useState(true);
+  const inputRef = useRef();
+  const deadlineRef = useRef();
+  // async function fetchAPI()
+  const fetchAPI = async () => {
+    setLoading(true);
+    try {
+      const response = await fetch(
+        "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+      );
+      const data = await response.json();
+      if (data) {
+        setTodos(data);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    // setLoading(true);
+    // fetch('https://jsonplaceholder.typicode.com/todos')
+    //   .then(response => response.json())
+    //   .then(data => setTodos(data))
+    //   .catch(error => console.error(error));
+    // setLoading(false);
+    fetchAPI();
+  }, []);
+
+  // Handle comleted to do
+  const handlecomleted = (index) => {
+    const newTodos = [...todos];
+    newTodos[index].completed = !newTodos[index].completed;
+    setTodos(newTodos);
+  };
+  // Add ToDo function
+  const handleAdd = () => {
+    const currentTodo = inputRef.current.value;
+    const currentDeadLineTodo = deadlineRef.current.value;
+
+    if (
+      currentTodo === "" ||
+      currentDeadLineTodo === "" ||
+      currentDeadLineTodo.toDateString < new Date().toDateString()
+    ) {
+      alert("Please enter a valid todo or DeadLine");
+      return;
+    }
+    const newElement = {
+      id: Date.now(),
+      description: currentTodo,
+      deadline: currentDeadLineTodo,
+      completed: false,
+    };
+    console.log(currentDeadLineTodo.toDateString < new Date().toDateString());

```

Comment:
We use `console.log()` for debugging purposes usually, so after you have finished debugging will be recommended to remove them.